install-dev/upgrade/php/migrate_tabs_17.php

```diff
@@ -59,12 +60,46 @@ function migrate_tabs_17()
     /* insert the new structure */
     ProfileCore::resetCacheAccesses();
     LanguageCore::resetCache();
-    $install = new Install();
-    $install->populateDatabase('tab');
+    if (!populateTab()) {
+        return false;
+    }
 
     /* update remaining idParent */
     foreach($moduleParents as $idParent => $className) {
         $idTab = Db::getInstance()->getValue('SELECT id_tab FROM '._DB_PREFIX_.'tab WHERE class_name='.pSQL($className));
         Db::getInstance()->execute('UPDATE '._DB_PREFIX_.'tab SET id_parent='.(int)$idTab.' WHERE id_parent='.(int)$idParent);
     }
+
+    return true;
+}
+
+function populateTab()
+{
+    $languages = [];
+    foreach (Language::getLanguages() as $lang) {
+        $languages[$lang['id_lang']] = $lang['iso_code'];
+    }
+
+    $xml_loader = new \XmlLoader1700();
+    $xml_loader->setTranslator(Context::getContext()->getTranslator());
+    $xml_loader->setLanguages($languages);
+
+    try {
+        $xml_loader->populateEntity('tab');
+    } catch (PrestashopInstallerException $e) {
+        return false;
+    }
+
+    return true;
+}
+
+class XmlLoader1700 extends XmlLoader
+{
+    public function createEntityTab($identifier, array $data, array $data_lang): void
```

Comment:
```suggestion
    public function createEntityTab($identifier, array $data, array $data_lang)
```

Reply:
That's an override, not sure we should change the signature here.

Reply:
Agreed!